src/biokbase/narrative/tests/test_jobcomm.py

```diff
@@ -148,190 +169,231 @@ def setUp(self):
+            self.check_id_error(req_dict, err)
 
-    def check_batch_job__dne_test(self, request_type):
-        job_id = JOB_NOT_FOUND
-        req = make_comm_msg(request_type, job_id, False)
-        err = JobIDException(JOB_NOT_REG_ERR, job_id)
-        with self.assertRaisesRegex(type(err), str(err)):
-            self.jc._handle_comm_message(req)
-        self.check_error_message(req, request_type, {JOB_ID: job_id}, err)
+    def check_batch_id__dne_test(self, request_type):
+        req_dict = make_comm_msg(
+            request_type + "_batch", {JOB_ID: JOB_NOT_FOUND}, False
+        )
+        err = JobIDException(JOB_NOT_REG_ERR, JOB_NOT_FOUND)
+        self.check_id_error(req_dict, err)
 
-    def check_batch_job__not_batch_test(self, request_type):
-        job_id = BATCH_COMPLETED
-        req = make_comm_msg(request_type, job_id, False)
-        err = JobIDException(JOB_NOT_BATCH_ERR, job_id)
-        with self.assertRaisesRegex(type(err), str(err)):
-            self.jc._handle_comm_message(req)
-        self.check_error_message(req, request_type, {JOB_ID: job_id}, err)
+    def check_batch_id__not_batch_test(self, request_type):
+        req_dict = make_comm_msg(
+            request_type + "_batch", {JOB_ID: BATCH_COMPLETED}, False
+        )
+        err = JobIDException(JOB_NOT_BATCH_ERR, BATCH_COMPLETED)
+        self.check_id_error(req_dict, err)
 
     # ---------------------
     # Send comms methods
     # ---------------------
     def test_send_comm_msg_ok(self):
         self.jc.send_comm_message("some_msg", {"foo": "bar"})
         msg = self.jc._comm.last_message
-        self.assertDictEqual(
+        self.assertEqual(
             msg,
             {
-                "content": None,
-                "data": {"content": {"foo": "bar"}, "msg_type": "some_msg"},
+                "msg_type": "some_msg",
+                "content": {"foo": "bar"},
             },
         )
         self.jc._comm.clear_message_cache()
 
     def test_send_error_msg__JobRequest(self):
         req = make_comm_msg("bar", "aeaeae", True)
-        self.jc.send_error_message("foo", req, {"extra": "field"})
+        self.jc.send_error_message(req, {"extra": "field"})
         msg = self.jc._comm.last_message
         self.assertEqual(
             {
-                "msg_type": "foo",
+                "msg_type": ERROR,
                 "content": {
                     "source": "bar",
-                    JOB_ID: "aeaeae",
                     "extra": "field",
                     "raw_request": req.raw_request,
                 },
             },
-            msg["data"],
+            msg,
         )
 
     def test_send_error_msg__dict(self):
-        req = make_comm_msg("bar", "aeaeae", False)
-        self.jc.send_error_message("foo", req, {"extra": "field"})
+        req_dict = make_comm_msg("bar", "aeaeae", False)
+        self.jc.send_error_message(req_dict, {"extra": "field"})
```

Comment:
throughout this file, I have used `req_dict` where the third arg to `make_comm_msg` is False, and `req` for where it's true